frontend/readhistory.lua

```diff
@@ -50,59 +48,48 @@ local function buildEntry(input_time, input_file)
         end,
         callback = function()
             selectCallback(input_file)
-        end
+        end,
     }
 end
 
-local function fileFirstOrdering(l, r)
-    if l.file == r.file then
-        return l.time > r.time
-    else
-        return l.file < r.file
+function ReadHistory:getIndexByFile(item_file)
+    for i, v in ipairs(self.hist) do
+        if item_file == v.file then
+            return i
+        end
     end
 end
 
-local function timeFirstOrdering(l, r)
-    if l.time == r.time then
-        return l.file < r.file
-    else
-        return l.time > r.time
+function ReadHistory:getIndexByTime(item_time)
+    local hist_num = #self.hist
+    if hist_num == 0 then
```

Comment:
I don't know if it's just me (and my French), but when I see "num", I think about the "numero" (the _number of a thing_, its index). When I see "nb", it's obvious it's the _number of things_, a count, "nombre" in French.
English may have "number" used for both things, so I dunno :)

Reply:
> I don't know if it's just me (and my French), but when I see "num", I think about the "numero" (the number of a thing, its index).

Yes. :-) To me "numero" is just a funny way of saying number.